R/estimatr_iv_robust.R

```diff
@@ -71,6 +126,18 @@ iv_robust <- function(formula,
   data <- enquo(data)
   model_data <- clean_model_data(data = data, datargs)
 
+  if (is.null(model_data$instrument_matrix)) {
+    stop(
+      "Must specify a `formula` with both regressors and instruments. For ",
+      "example, `formula = y ~ x1 + x2 | x1 + z2` where x1 and x2 are the ",
+      "regressors and z1 and z2 are the instruments.\n\nSee ?iv_robust."
+    )
+  }
```

Comment:
Is there someway to write this error message so that it checks the formula (an input) rather than the output of clean model data.

Reply:
Sure, but this is what implicitly happens in clean_model_data, and I'm a little worried about making sure I'm evaluating the formula arg properly before testing it

Reply:
kk, suggest moving this error check into clean_model_data then. the output of clean_model_data should (if it succeeds) not need typically need further testing / checking.

Reply:
I don't know why I would be averse to internal arguments.... but I was. This makes sense.